lib/ex_doc/formatter/html/autolink.ex

```diff
@@ -120,6 +120,22 @@ defmodule ExDoc.Formatter.HTML.Autolink do
     typespec_to_string(other, typespecs, aliases, lib_dirs)
   end
 
+  defp normalize_left({:::, _, [{name, _, args}, right]}, typespecs, aliases, lib_dirs) do
+    placeholder = {:::, [], [{:_, [], args}, right]}
+    "_" <> rest = typespec_to_string(placeholder, typespecs, aliases, lib_dirs)
+    Atom.to_string(name) <> rest
```

Comment:
I am afraid though this breaks for operators. For example: `@type integer + integer :: integer` :(